CRM/Financial/BAO/FinancialItem.php

```diff
@@ -295,13 +295,28 @@ public static function checkContactPresent($contactIds, &$error) {
    * @return object CRM_Core_DAO
    */
   public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
+    $liabilityAccountType = CRM_Core_OptionGroup::getValue('financial_account_type', 'Liability', 'name');
     $queryParams = array(
       1 => array($entityId, 'Integer'),
       2 => array($entityTable, 'String'),
+      3 => array($liabilityAccountType, 'Integer'),
     );
-    $query = 'SELECT id, description, status_id, financial_account_id 
-      FROM civicrm_financial_item
-      WHERE entity_id = %1 AND entity_table = %2 ORDER BY id DESC LIMIT 1';
+    $query = 'SELECT cfi.id, cfi.description, cfi.amount, cfi.financial_account_id, cfi.status_id
```

Comment:
I don't understand why you're only choosing these fields. Don't you need to copy contact_id and currency to new financial_item? What about *